packages/compiler/core/types.ts

```diff
@@ -260,6 +260,11 @@ export interface Model extends BaseType, DecoratedType, TemplatedTypeBase {
    */
   sourceModel?: Model;
 
+  /**
+   * The models that this model is composed of via spread or intersection.
+   */
+  sourceModels?: Model[];
```

Comment:
I think we should find something else for this name, being just the plural of `sourceModel` will make it confusing.

Maybe one of:
- `composingModels`
- `composedByModels`
- `constituentModels`

Reply:
Agree on a different name. Not sure what's a good one. @bterlson any suggestions?

Reply:
Also, should there be a separate property for composed via intersection vs. composed via spread?

Reply:
Right now those 2 should behave the exact same but we do have this issue about making potentially different semantic for `...` which would override previous values vs `&` which conflict/merge https://github.com/microsoft/typespec/issues/1292

With that in mind maybe we want to differentiate or at least leave room to add a non breaking change difference later

Reply:
I guess spreadModels and intersectedModels solves the naming issue and future proofs for a difference in behavior, with the trade-off that you might look at only one or the other when you should look at both.